aten/src/ATen/SparseTensorUtils.h

```diff
@@ -128,4 +129,37 @@ inline LongTensor flatten_indices_by_dims(const LongTensor& indices, const IntAr
   return new_indices;
 }
 
+inline LongTensor coo_to_csr(const int64_t* indices, int64_t dim, int64_t nnz) {
+  /* 
+    Find the CSR representation for a row `indices` from the COO format 
+    Inputs:
+      `indices` is the row pointer from COO indices
+      `dim` is the row dimensionality 
+      `nnz` is the number of non-zeros 
+    
+    Output: 
+      `csr` is a compressed row array in a CSR format
+  */
+  LongTensor csr = native::zeros({dim + 1}, kLong);
+
+  // TODO: eliminate this conditional when zero-size dims supported correctly
+  if (nnz > 0) {
+    auto csr_accessor = csr.accessor<int64_t, 1>();
+    // Convert the sparse matrix to CSR format
+    at::parallel_for(0, nnz, 10000, [&](int64_t start, int64_t end) {
```

Comment:
Why the 10,000 constant here?

Reply:
I just move this function which was already defined at `SparseTensorMath.cpp` as now is used for both sparse-dense and sparse-sparse matmul. I think that 10, 000 was used here to choose and `optimal` `num_threads` at `at::parallel_for`

```
if (grain_size > 0) {
   num_threads = std::min(num_threads, divup((end - begin), grain_size));
}
```